src/ObjCRuntime/Blocks.cs

```diff
@@ -83,14 +83,19 @@ public unsafe struct BlockLiteral {
 		[DllImport ("__Internal")]
 		static extern IntPtr xamarin_get_block_descriptor ();
 
+		struct DelegatePairs {
+			public Delegate Trampoline, UserDelegate;
+		}
+
 		//
 		// trampoline must be static, and someone else needs to keep a ref to it
 		//
 		public unsafe void SetupBlock (Delegate trampoline, Delegate userDelegate)
 		{
 			isa = block_class;
 			invoke = Marshal.GetFunctionPointerForDelegate (trampoline);
-			local_handle = (IntPtr) GCHandle.Alloc (userDelegate);
+			var delegates = new DelegatePairs () { Trampoline = trampoline, UserDelegate = userDelegate };
```

Comment:
I am not sure that there is much of to be saved here, the extra cost with the struct is the size of a pointer, adding more complexity for something that is saving 8 bytes on 64 bit platform is not worth it IMO.

Using a Tuple<> would use more memory, because it would require an additional allocation, and it is an object, so some 40-something bytes?

What do you mean that there is a fix needed in line 160?